helper/resource/testing.go

```diff
@@ -22,6 +22,12 @@ import (
 
 const TestEnvVar = "TF_ACC"
 
+// TestProvider is implemented by scheme.Provider to allow resetting the
+// provider state between tests.
```

Comment:
Nitpick but I'd reword the comment to say that this can be implemented by any ResourceProvider to have custom reset functionality, and note when TestReset is called. You can also make a note that helper/schema implements this.